src/workspace-data/Data.js

```diff
@@ -593,37 +593,43 @@ export const WorkspaceData = _.flow(
     }, [workspaceId, wdsApp]);
 
     const loadEntityMetadata = async () => {
-      try {
-        setEntityMetadata(undefined);
-        setEntityMetadataError(false);
-        const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata();
+      if (isGoogleWorkspace) {
+        try {
+          setEntityMetadata(undefined);
+          setEntityMetadataError(false);
+          const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata();
 
-        if (selectedData?.type === workspaceDataTypes.entities && !entityMetadata[selectedData.entityType]) {
+          if (selectedData?.type === workspaceDataTypes.entities && !entityMetadata[selectedData.entityType]) {
+            setSelectedData(undefined);
+          }
+          setEntityMetadata(entityMetadata);
+        } catch (error) {
+          reportError('Error loading workspace entity data', error);
+          setEntityMetadataError(true);
           setSelectedData(undefined);
+          setEntityMetadata({});
         }
-        setEntityMetadata(entityMetadata);
-      } catch (error) {
-        reportError('Error loading workspace entity data', error);
-        setEntityMetadataError(true);
-        setSelectedData(undefined);
-        setEntityMetadata({});
+      } else {
+        setEntityMetadata({}); // This is not used for AzureWorkspaces
       }
     };
 
     const loadSnapshotMetadata = async () => {
       try {
         setSnapshotMetadataError(false);
-        const { gcpDataRepoSnapshots: snapshotBody } = await Ajax(signal).Workspaces.workspace(namespace, name).listSnapshots(1000, 0);
+        if (isGoogleWorkspace) {
```

Comment:
instead of 
```javascript
if (isGoogleWorkspace) {
    … lots of stuff …
}
```
what about:
```javascript
if (isAzureWorkspace) {
  return;
}
… lots of stuff …
```
then, you don't have everything within a long closure

Reply:
So clever!